servers/src/main/java/tachyon/master/file/FileSystemMaster.java

```diff
@@ -463,13 +472,18 @@ public long createFile(TachyonURI path, long blockSizeBytes, boolean recursive)
       List<Inode> created =
           createFileInternal(path, blockSizeBytes, recursive, System.currentTimeMillis());
 
-      // Journal all of the created inodes.
-      for (Inode inode : created) {
-        writeJournalEntry(inode.toJournalEntry());
-      }
       writeJournalEntry(mDirectoryIdGenerator.toJournalEntry());
+      for (int i = 0; i < created.size(); i ++) {
+        Inode inode = created.get(i);
+        if (i == 0) {
+          // The first inode in the list was modified, not created.
```

Comment:
Using a separate data structure (or a Pair) does make sense as it helps to call out the difference between the first one (modified) and the rest (created).